### ISSUE TITLE
How to speed up differentiable rendering？

### ISSUE THREAD

**Issue by zhangzscn** [facebookresearch/pytorch3d#951]
Right now I need Silhouette and Zbuf to calculate the loss, but feels like rendering is slow.

I have tried to adjust the Faces_per_pixel to less than 100, like 50, but this will affect the Silhouette accuracy. If the faces_per_pixel is 100, the Silhouette range is [0,1], and if the faces_per_pixel is 50, the Silhouette range is [0,1), most of them will be less than 1.

How can I speed up rendering by adjusting Settings while keeping the results accurate？

<details>
<summary>this is my code for rendering</summary>

```
class DiffRenderer(nn.Module):
    def __init__(self,
                 camera_parameter: CameraParameter,
                 device: Device = "cpu"):
        super(DiffRenderer, self).__init__()

        blend_params = BlendParams(sigma=1e-4, gamma=1e-4)
        image_size = camera_parameter.image_size_list.cpu().detach().numpy().astype(numpy.int).tolist()[0]

        self.cameras = PerspectiveCameras(focal_length=camera_parameter.focal_length_list,
                                          principal_point=camera_parameter.principal_point_list,
                                          image_size=camera_parameter.image_size_list,
                                          in_ndc=True)

        raster_settings = RasterizationSettings(image_size=image_size,
                                                blur_radius=np.log(
                                                    1. / 1e-4 - 1.) * blend_params.sigma,
                                                faces_per_pixel=100,
                                                perspective_correct=False)

        self.renderer = MeshRenderer(rasterizer=MeshRasterizer(cameras=self.cameras,
                                                               raster_settings=raster_settings),
                                     shader=SoftSilhouetteShader(blend_params=blend_params))

        self.renderer.to(device)

        raster_settings = RasterizationSettings(image_size=image_size,
                                                blur_radius=0,
                                                faces_per_pixel=1,
                                                perspective_correct=False)

        self.mesh_rasterizer = MeshRasterizer(cameras=self.cameras,
                                              raster_settings=raster_settings)

    def forward(self, meshes, r, t):
        images = self.renderer(meshes_world=meshes, R=r, T=t)
        fragments = self.mesh_rasterizer(meshes_world=meshes, R=r, T=t)

        return images[..., 3], fragments.zbuf[..., 0]
```
</details>

**Comment by bottler** [comment]
Are you using a GPU? How long does the rendering take? How big is the problem (e.g. image size, number of verts and faces)?

Some ideas: Can you profile the code (with and without CUDA_LAUNCH_BLOCKING) to see where the time is spent? Can you experiment with the bin_size in the RasterizationSettings?
